pkg/users/git.go

```diff
@@ -0,0 +1,161 @@
+package users
+
+import (
+	"fmt"
+
+	jenkinsv1 "github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	"github.com/jenkins-x/jx/pkg/log"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	jenkninsv1client "github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+
+	"github.com/jenkins-x/jx/pkg/gits"
+)
+
+// GitUserResolver allows git users to be converted to Jenkins X users
+type GitUserResolver struct {
+	GitProvider gits.GitProvider
+	JXClient    jenkninsv1client.Interface
+	Namespace   string
+}
+
+// GitUserAsUser will convert the GitUser to a Jenkins X user and attempt to complete the user info by:
+// * checking the user custom resources to see if the user is present there
+// * making a call to the gitProvider
+// as often user info is not complete in a git response
+func (r *GitUserResolver) GitUserAsUser(user *gits.GitUser) (*jenkinsv1.User, error) {
```

Comment:
It would be nice to have a test for this method since has a lot of complex logic. You can use the fake client.